server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -464,6 +501,52 @@ tableName, getSerialType(), getPayloadType()
     );
   }
 
+  /**
+   * Adds the used_flag_last_updated column to the Druid segment table.
+   *
+   * This is public due to allow the UpdateTables cli tool to use for upgrade prep.
+   */
+  @Override
+  public void alterSegmentTableAddUsedFlagLastUpdated()
+  {
+    String tableName = tablesConfigSupplier.get().getSegmentsTable();
+    if (!tableHasColumn(tableName, "used_flag_last_updated")) {
```

Comment:
thinking out loud: should this column check be put into alterTable instead of this location?